opm/autodiff/RedistributeDataHandles.hpp

```diff
@@ -40,12 +40,84 @@ inline void distributeGridAndData( Grid& ,
                                    BlackoilPropsAdFromDeck& ,
                                    DerivedGeology&,
                                    std::shared_ptr<BlackoilPropsAdFromDeck::MaterialLawManager>&,
+                                   std::vector<double>&,
                                    boost::any& ,
                                    const bool )
 {
 }
 
 #if HAVE_DUNE_CORNERPOINT && HAVE_MPI
+/// \brief a data handle to distribute the threshold pressures
+class ThresholdPressureDataHandle
```

Comment:
On Thu, Feb 18, 2016 at 09:14:23AM -0800, Atgeirr Flø Rasmussen wrote:
> This class seems to me to work on a cell level, and then for each cell communicating the threshold pressures for each adjacent face. That would communicate almost twice as many values as necessary. Could that be reduced by communicating on faces instead?
> 

Yes it does and yes it could with some work effort.

As most of the communication takes place on cells, one
can currently only communicate on cells. Adding communication to the
interfaces is possible (at least for newer DUNE versions), but we have
to store an additional communication pattern. Maybe we could throw that
away after this communication step, though.